source/auth/auth.rst

```diff
@@ -468,7 +468,7 @@ mechanism_properties
 Hostname Canonicalization
 `````````````````````````
 
-If CANONICALIZE_HOST_NAME is true, the client MUST canonicalize the name of each host it uses for authentication. There are two options. First, if the client's underlying GSSAPI library provides hostname canonicalization, the client MAY rely on it. For example, MIT Kerberos has `a configuration option for canonicalization <https://web.mit.edu/kerberos/krb5-1.13/doc/admin/princ_dns.html#service-principal-canonicalization>`_.
+If CANONICALIZE_HOST_NAME is true, "forward", or "forwardAndReverse", the client MUST canonicalize the name of each host it uses for authentication. There are two options. First, if the client's underlying GSSAPI library provides hostname canonicalization, the client MAY rely on it. For example, MIT Kerberos has `a configuration option for canonicalization <https://web.mit.edu/kerberos/krb5-1.13/doc/admin/princ_dns.html#service-principal-canonicalization>`_.
```

Comment:
What are the valid values for `CANONICALIZE_HOST_NAME`. Based on this statement, valid values include `true`, `"forward"`, and `"forwardAndReverse"`. I assume `false` would also be valid. In the spec tests below, `"none"` is also used.

Should true/false be deprecated? If not, should `false` or `"none"` be used when disabling canonicalization.

What is the expected behaviour for an unknown value like `"off"` or `"on"`? Should an error be raised? Should it default to true? Or false? My concern here is what if someone uses `"forwardReverse"` rather than `"forwardAndReverse"` by mistake? Or if they type `"foward"` (misspelling)?

We should clearly enumerate the allowed values for `CANONICALIZE_HOST_NAMES` and specify whether an error or warning should be raised for unknown values.

Reply:
I've called out now all valid values in the spec for `CANONICALIZE_HOST_NAME` and specified that drivers MUST error if the value provided is not one of them. I've decided not to deprecate `true` and `false` and just say that drivers MAY allow them, since I figured many apps may depend on `true` and `false` and not want to change them as these changes were driven by the dev tools team.